src/main/java/org/opensearch/geospatial/ip2geo/action/GetDatasourceResponse.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.action;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.action.ActionResponse;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.geospatial.ip2geo.jobscheduler.Datasource;
+
+/**
+ * Ip2Geo datasource get request
+ */
+@Getter
+@Setter
+@Log4j2
+@EqualsAndHashCode
+public class GetDatasourceResponse extends ActionResponse implements ToXContentObject {
+    private static final String FIELD_NAME_DATASOURCES = "datasources";
+    private static final String FIELD_NAME_NAME = "name";
+    private static final String FIELD_NAME_STATE = "state";
+    private static final String FIELD_NAME_ENDPOINT = "endpoint";
+    private static final String FIELD_NAME_UPDATE_INTERVAL = "update_interval_in_days";
+    private static final String FIELD_NAME_NEXT_UPDATE_AT = "next_update_at_in_epoch_millis";
+    private static final String FIELD_NAME_NEXT_UPDATE_AT_READABLE = "next_update_at";
+    private static final String FIELD_NAME_DATABASE = "database";
+    private static final String FIELD_NAME_UPDATE_STATS = "update_stats";
```

Comment:
As best practice we use something called as ParseField to define the field names that used in parsing, Lets use that. Its a great abstraction over these private static final strings

Reply:
We don't parse anything here. Just used for xContent generation.

Reply:
Still better if we use parse field.

Reply:
Done